collector_test.go

```diff
@@ -6,9 +6,90 @@ import (
 
 	"github.com/soniah/gosnmp"
 
+	"github.com/prometheus/client_model/go"
 	"github.com/prometheus/snmp_exporter/config"
 )
 
+func TestPduToSample(t *testing.T) {
+	cases := []struct {
+		pdu            *gosnmp.SnmpPDU
+		indexOids      []int
+		metric         *config.Metric
+		oidToPdu       map[string]gosnmp.SnmpPDU
+		expectedMetric string
+	}{
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Type:  0x02,
+				Value: 2,
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.1.1.1.1",
+				Type: "counter",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedMetric: "counter:<value:2 > ",
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Type:  0x02,
```

Comment:
Use the names here for clarity: https://github.com/soniah/gosnmp/blob/adbb107aeaa6c23e71056bfd7a095874a07c7d04/gosnmp.go#L134